Fix wrong behavior of Eq & Ord

addap opened the pull request:
I realized that the easiest way is to just rely on the derived implementations of the Eq and Ord traits. 
To still get alphabetical sorting we must put the name field before the id field.
Added two tests that fail on master and pass now.

addap committed:
Add tests for wrong behavior of Eq & Ord

As described in #35

addap committed:
Use derived Eq & Ord for Type

The previous handwritten implementations were not consistent with each
other as they allowed values a & b where a == b but a < b.
Fixes #35

hirrolot reviewed (changes requested):
Thanks, LGTM. Would you also add a `CHANGELOG.md` entry explaining your changes? Thanks!

addap committed:
Use rustfmt & reword comment

addap committed:
Add test for PartialEq for DependencyMap

The current implementation only zips the two key iterators together. So if
one of them is a subset of the other it only checks the keys in the
subset for equality. In the simplest case, the empty dependency map is
equal to every other dependency map.

addap committed:
Fix PartialEq implementation of DependencyMap

Analogous to how equality is defined for BTreeMap, but only looking at keys.

addap committed:
Add changelog entry

hirrolot reviewed (approved):
Thanks again for figuring this out!

addap commented:
Okay, I just added an entry to the Changelog. 

Also I was looking at other PartialEq implementations in the project and noticed that the one for `DependencyMap` might be wrong. When zipping together the two lists, if one of them is shorter than the other, the zip ends after the shorter list does not have any more items. As a consequence, only keys in the shorter list are checked for equality and therefore the empty dependency map is equal to every other. I assume this is wrong so I added a test & fix for that, too.

hirrolot commented:
I've release v0.5.1 with your fix.
